pkg/kubelet/images/image_manager.go

```diff
@@ -172,13 +170,11 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 // applyDefaultImageTag parses a docker image string, if it doesn't contain any tag or digest,
 // a default tag will be applied.
 func applyDefaultImageTag(image string) (string, error) {
-	named, err := dockerref.ParseNormalizedNamed(image)
```

Comment:
@bart0sh This is the best I think we can do.  

I changed this code to use ParseImageName and use that to populate the tag.  This seems to be used in more places.  I don't think we can remove the logic for checking ImageNames but we can atleast standarize on a way to ParseImageNames.

Reply:
Why changing this to `ParseImageName`? It does the same thing right?

Reply:
And they are similar.  Both check the same function from docker distribution but their returns are different.

Reply:
gotcha. Would this cause problems for images with `latest` tag already? `nginx:latest` will become `nginx:latest:latest` right?

Reply:
good call out. I'll add a test case for that.

Reply:
Added.